pkg/operator/operator.go

```diff
@@ -68,57 +65,17 @@ func New(config operatorconfig.Config) (*Operator, error) {
 		return nil, fmt.Errorf("failed to create operator controller: %v", err)
 	}
 
-	// Create additional controller event sources from informers in the managed
-	// namespace. Any new managed resources outside the operator's namespace
-	// should be added here.
-	mapper, err := apiutil.NewDiscoveryRESTMapper(kubeConfig)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get API Group-Resources")
-	}
-	operandCache, err := cache.New(kubeConfig, cache.Options{Namespace: "openshift-dns", Scheme: scheme, Mapper: mapper})
-	if err != nil {
-		return nil, fmt.Errorf("failed to create openshift-dns cache: %v", err)
-	}
-	// Any types added to the list here will only queue a cluster dns if the
-	// resource has the expected label associating the resource with a cluster dns.
-	for _, o := range []runtime.Object{
-		&appsv1.DaemonSet{},
-		&corev1.Service{},
-		&corev1.ConfigMap{},
-	} {
-		// TODO: may not be necessary to copy, but erring on the side of caution for
-		// now given we're in a loop.
-		obj := o.DeepCopyObject()
-		informer, err := operandCache.GetInformer(obj)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get informer for %v: %v", obj, err)
-		}
-		operatorController.Watch(&source.Informer{Informer: informer}, &handler.EnqueueRequestsFromMapFunc{
-			ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
-				labels := a.Meta.GetLabels()
-				if dnsName, ok := labels[manifests.OwningClusterDNSLabel]; ok {
-					logrus.Infof("queueing cluster dns %s, related: %s", dnsName, a.Meta.GetSelfLink())
-					return []reconcile.Request{
-						{
-							NamespacedName: types.NamespacedName{
-								Name: dnsName,
-							},
-						},
-					}
-				} else {
-					return []reconcile.Request{}
-				}
-			}),
-		})
-	}
+	operatorController.Watch(&source.Kind{Type: &operatorv1.DNS{}}, &handler.EnqueueRequestForObject{})
+	operatorController.Watch(&source.Kind{Type: &appsv1.DaemonSet{}}, &handler.EnqueueRequestForOwner{})
+	operatorController.Watch(&source.Kind{Type: &corev1.Service{}}, &handler.EnqueueRequestForOwner{})
+	operatorController.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForOwner{})
```

Comment:
This has been nagging me for a while: Shouldn't these watches be established in `operatorcontroller.New`?

Reply:
Fixed